vyper/parser/expr.py

```diff
@@ -292,21 +292,22 @@ def variables(self):
     def attribute(self):
         # x.balance: balance of address x
         if self.expr.attr == 'balance':
-            if getattr(self.expr.value, 'id', None) == "self" and version_check(begin="istanbul"):
-                return LLLnode.from_list(
-                    ['selfbalance'],
-                    typ=BaseType('uint256', {'wei': 1}),
-                    location=None,
-                    pos=getpos(self.expr),
-                )
             addr = Expr.parse_value_expr(self.expr.value, self.context)
             if not is_base_type(addr.typ, 'address'):
                 raise TypeMismatchException(
                     "Type mismatch: balance keyword expects an address as input",
                     self.expr
                 )
+            if (
+                isinstance(self.expr.value, ast.Name) and
+                self.expr.value.id == "self" and
```

Comment:
Wouldn't `addr` be able to be used directly here now?

Reply:
I don't know how to get from the parsed expr to `self`.  Not saying it's not possible, but I can't figure out how :grimacing:

Reply:
Eh, I'm just being anal